tests/data/test_binary_performance.py

```diff
@@ -247,3 +247,52 @@ def test_per_context_does_not_leak_state_across_iterations(self, fake_seismo):
 
         # Combining scores for each (score, target) pair should have two positives
         assert result["Positives"].tolist() == [2, 2, 2, 2]
+
+    def test_generate_analytics_data_metric_differs_but_is_close(self, fake_seismo):
+        fake_seismo.event_aggregation_method = lambda target: "max"
+
+        rng = np.random.default_rng(42)
+        y_true = rng.integers(0, 2, size=1000)  # Balanced 0s and 1s
+        y_score = rng.uniform(0, 1, size=1000)  # Uniform prediction scores between 0 and 1
+        df = pd.DataFrame(
+            {
+                "target1": y_true,
+                "score1": y_score,
+            }
+        )
+
+        fake_seismo.dataframe = df
+
+        result_low = generate_analytics_data(
+            score_columns=["score1"],
+            target_columns=["target1"],
+            metric="Sensitivity",
+            metric_values=[0.5],
+            decimals=3,
+            censor_threshold=1,
+        )
+        result_high = generate_analytics_data(
+            score_columns=["score1"],
+            target_columns=["target1"],
+            metric="Sensitivity",
+            metric_values=[0.5],
+            decimals=5,
+            censor_threshold=1,
+        )
+
+        assert result_low is not None and result_high is not None
+
+        # Ensure both versions yield same structure
+        assert list(result_low.columns) == list(result_high.columns)
+
+        for col, atol in [("0.5_Threshold", 0.1), ("AUROC", 0.001)]:
+            assert col in result_low.columns and col in result_high.columns
+
+            val_low = result_low[col].iloc[0]
+            val_high = result_high[col].iloc[0]
+
+            # Values differ (better precision affects threshold resolution)
+            assert val_low != val_high
+
```

Comment:
```suggestion
            # Values differ (better precision affects threshold resolution)
            assert val_low != val_high, "These can collide with probability 1:100"
```
Really since we have a for loop, the probability that this fails is `1 - (.99)^2` which is close to 1:50
Can we not use RNG?

If we want to use RNG and we want to automatically rerun on failure, there are some plugins for this:
https://docs.pytest.org/en/stable/explanation/flaky.html has some suggestions if we wanted to go that route (https://github.com/pytest-dev/pytest-rerunfailures?tab=readme-ov-file#re-run-individual-failures)

I don't think that extra work is needed, and an assertion comment should suffice.

Reply:
We could also drop the assert that they are not equal.

Reply:
I didn't catch the seed of `42` up top in `rng = np.random.default_rng(42)`. 

I still think an assertion comment would be useful. Something like 

```suggestion
            # Values differ (better precision affects threshold resolution)
            assert val_low != val_high, "Differing precision expected, test seed was chosen to verify these two values are not equal"

```